woocommerce/woocommerce-gutenberg-products-block
# Fix: Make filter block contextual - Editor

## What

This PR adds an utility to be used for Edit components of filter blocks. The utility is the JS equivalent of the PHP one implemented in #10919.

Fixes #10985 

## Why

Unlike the #10919, we don't passing any data in this PR, we just provide an utility for filter blocks to consume. The Product Collection query is already available as context.



## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. On a new page, Add Product Collection block.
2. Add a Collection Filters block inside the Product Collection block.
3. Add Collection Price Filter block inside Collection Filters.
4. See the block shows correct price range.
5. Adjust the setting of Product Collection block to change the min or max price range of products.
6. See the Collection Price Filter block reacts and updates the price range accordingly.
7. Save the page and see changes applied on the front end.


## Screenshots or screencast

Uploading Screen Recording 2023-10-12 at 13.01.25.mov…

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Add suggested changelog entry here.
